pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -415,6 +415,21 @@ func (as *availabilitySet) GetZoneByNodeName(name string) (cloudprovider.Zone, e
 		return cloudprovider.Zone{}, err
 	}
 
+	// Get availability zone for the node.
+	if vm.Zones != nil && len(*vm.Zones) > 0 {
+		zones := *vm.Zones
+		zoneID, err := strconv.Atoi(zones[0])
```

Comment:
Is it possible that a VM has multiple zones?

Reply:
No, a VM must be in one exact zone (while there could multiple zones in scaleset itself).